lib/mongo/server/connectable.rb

```diff
@@ -103,7 +107,9 @@ def ensure_same_process!
       end
 
       def read
-        ensure_connected{ |socket| Protocol::Reply.deserialize(socket) }
+        read_with_retry do
```

Comment:
This will not work as the connection does not know anything about the cluster and the retry code calls `cluster.scan!`.